### PR TITLE
[Wallet] Add Portuguese

### LINKED ISSUE #5949: Portuguese language support
### Expected Behavior

App should be available in Portuguese.

### Current Behavior

Portuguese is not available as a language.

### PR DESCRIPTION
**Pull request by jeanregisser**
### Description

Portuguese is here 🇧🇷!

### Other changes

- Lazy load locale data (only the data for the currently selected locale is loaded in memory)
- Only include the `date-fns` locales we actually use in the final bundle. Bundle size went from **9.6MB** to **8.98MB** (~6.45% smaller)!

**bundle size before / after** (using `npx visualize-bundle`)
Screenshot 2020-12-22 at 17 14 01Screenshot 2020-12-22 at 17 07 15

- Centralize all locale data config in `locales/index.ts` (name, strings and date-fns locale) so we don't need to modify different places in the code when adding a new language.
- Fix whole translation strings logged during i18n init (debug mode was on for some reason). Hopefully this should help app startup performance.
- Remove isolated translations for currencies in `base` project.
  - It's fine to have some repetition and gives better context to the translator.
  - Also the current solution didn't work well with different plural rules.
- Add simple test for i18n.ts

### Related issues

- Fixes #5949

### Backwards compatibility

Yes

### CODE REVIEW

File: packages/mobile/ios/celo/es-419.lproj/InfoPlist.strings [hunk 1]
```diff
@@ -0,0 +1,3 @@
+NSCameraUsageDescription = "Celo Wallet needs camera access to scan QR codes.";
```
Comment (gnardini) [inline packages/mobile/ios/celo/es-419.lproj/InfoPlist.strings]: Should these be translated? They also say "Celo Wallet", I think you started a thread on Slack about changing it recently, right?

**Review by gnardini — APPROVED** [review]
Amazing to have another language on the app 🎉 

Quick question, how have people been doing to add spanish translations? In the future, I assume I need to add portuguese translations for new strings, right? Do we have any process for this or do people just use google translate?

File: packages/mobile/ios/celo/es-419.lproj/InfoPlist.strings (hunk 1 above)
Comment (jeanregisser) [reply packages/mobile/ios/celo/es-419.lproj/InfoPlist.strings]: Yes they should 👍 
I'm waiting to add them, should happen today.

**Comment by jeanregisser** [comment]
> Quick question, how have people been doing to add spanish translations? In the future, I assume I need to add portuguese translations for new strings, right? Do we have any process for this or do people just use google translate?

We want to automate this and have engineers only worry about the English strings.
Then during the release process we would integrate the final strings in the other languages we support. 
We're not there yet, and are still evaluating the process with the translation agency we're working with.

**Comment by Lss-Ankit** [comment]
Hi @gnardini , @jeanregisser I have verified with Android Internal Build V1.8.0(1004294330) & Test Flight Build V1.8.0(40) and observed that we are able to see Portuguese(Brasil) language in our application. And also we are able to convert app language in Portuguese(Brasil).
**Attachment:** [PortuguesLanguage.png](https://drive.google.com/file/d/1PO8vkJJWsUBMQNhtXStPmyWelRRM3eim/view?usp=sharing)
Also we have observed following observations:

1. UI of Request , Set Up Now button text get cut on Samsung Galaxy A5(7.0) device when user change language to Portuguese(Brasil) language.
**Attachment:** [UIOfRequestAndConfirmNowButton.png](https://drive.google.com/file/d/1o2_LGEOE4Xi0lcbsyg_2uTUSvZSMxf0u/view?usp=sharing)

2. Also it is observed that sometimes application Portugues(Brasil) language & google translator language not seems same.
- For Example: For send funds across borders walkthrough screen google shows " enviar fundos através das fronteiras " and in our application shows  "enviar fundos internacionalmente". Some similar words are displayed instead of correct words
Let me know needful.
**Verified on Devices:** Samsung Galaxy A5(7.0), iPhone XR(13.3.1)

3. FYI: We have raised a new ticket https://github.com/celo-org/celo-monorepo/issues/6444 for UI related issues as user changes language.

**Pull request merged**